### ISSUE TITLE
Defining Example objects

### ISSUE THREAD

**Issue by vdecristofaro** [NSwag/NSwag#316]
Is there any way to generate example objects of operation responses, as foreseen by the [specification](http://swagger.io/specification/#exampleObject)?

**Comment by rsuter** [comment]
This scenario is currently not supported.

**Comment by rsuter** [comment]
Is this a duplicate of https://github.com/NSwag/NSwag/issues/47?

**Comment by vdecristofaro** [comment]
Yes it is. 
The scope is to find a way to create request/responses samples to be used in automated creation of api documentation.

**Comment by rsuter** [comment]
Can you post a sample? i.e. a sample of an expected Swagger spec?

**Comment by vdecristofaro** [comment]
What do you mean by sample?

I would like to create object samples and specify them statically at the method level. One option could be to extend the `ResponseType `attribute to receive on abjiect which specifies the mime type and and object sample. For example, instead of this 

`[ResponseType( "200", typeof( BaseServiceResult ), Description = "Success" )]
`
it can be done like this

```
[ResponseType( "200", typeof( BaseServiceResult ), Description = "Success", 
                          new ExampleObject( "application/json", new { 
                             propertyA = "",
                             propertyB = 123
                          } ) )]
```

However this is not the best solution at all because you need to specify an object for every `ResponseType`. In my case I can live with this

**Comment by rsuter** [comment]
I think you cannot create a new instance of an own class in an attribute (limitation in C#)...

**Comment by rsuter** [comment]
See

- http://swagger.io/specification/#responseObject
- http://swagger.io/specification/#exampleObject

**Comment by vdecristofaro** [comment]
> I think you cannot create a new instance of an own class in an attribute (limitation in C#)...

Right. What about passing a stringified json to be parsed internally?

**Comment by rsuter** [comment]
Yes, this would work...

**Comment by juliencousineau** [comment]
+ 1

**Comment by RSuter** [comment]
Maybe this solves your problem?

https://github.com/RSuter/NSwag/blob/master/src/NSwag.SwaggerGeneration.WebApi.Tests/OperationProcessors/SwaggerOperationProcessorAttributeTests.cs

**Comment by vdecristofaro** [comment]
Could you please add a sample usage?

**Comment by RSuter** [comment]
https://blog.rsuter.com/nswag-tutorial-implement-a-custom-operation-processor-to-define-redoc-code-samples/

**Comment by comdw** [comment]
I had the same requirement and am currently comparing and deciding on whether to use Swashbuckle or NSwag.  There is a way to do this in Swashbuckle described [here](https://github.com/domaindrivendev/Swashbuckle/issues/162), don't know if that's any help here.

Contrary to the comment about this not changing anything in the Swagger UI, it actually did also render my example response under the "Example Value" section of Response Class.

The generated swagger specification then looks something like this:
```
"definitions": {
	"User": {
		"description": "External representation of a User",
		"type": "object",
		"properties": {
			"name": {
				"description": "User name.",
				"type": "string"
			},
			"language": {
				"description": "User's language.",
				"type": "string"
			},
			"age": {
				"description": "user's age.",
				"type": "int"
			}
		},
		"example": {
			"name": "Bob",
			"language": "en-GB",
			"age": 36,
		}
	}
}
```

**Comment by RSuter** [comment]
You can also implement a custom processor for NSwag: https://blog.rsuter.com/nswag-tutorial-implement-a-custom-operation-processor-to-define-redoc-code-samples/

**Comment by comdw** [comment]
Does that work for SwaggerUI as well as redoc?

**Comment by RSuter** [comment]
I think `x-code-samples` is specific to ReDoc, but you can add any additional properties with a custom processor - you have to check the Swagger UI documentation what properties are supported...

**Comment by RSuter** [comment]
Ref: https://stackoverflow.com/questions/50466680/how-can-i-generate-a-custom-response-example-value-in-nswag-asp-net-core

**Comment by softlion** [comment]
Why not extract the <example> tag like it is done for the <summary> tag on each of the property ?
